src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -112,13 +115,15 @@ static bool templateNotify
     cer.contextElement = ce;
     ncr.subscriptionId = subscriptionId;
     ncr.contextElementResponseVector.push_back(&cer);
-    payload = ncr.toJson(renderFormat, attrsOrder);
+    payload  = ncr.toJson(renderFormat, attrsOrder);
+    mimeType = "application/json";
   }
   else
   {
     macroSubstitute(&payload, httpInfo.payload, ce);
+    payload      = curl_unescape(payload.c_str(), payload.length());
     renderFormat = NGSI_V2_CUSTOM;
-    LM_W(("KZ: substituted payload: '%s'", payload.c_str()));
+    mimeType     = "text/plain; charset:utf-8";  // May be overridden by 'Content-Type' in 'headers'
```

Comment:
To be homogeneous with L119 for the JSON case, I'd suggest to use `mimeType = "texp/plain";` here, then add to the `if` in httpSendRequest() (or something like that, don't remembering the exact name of the function right now) `orig_content_type == "application/text"` to sum the charset part.

By the way, is `charset=utf-8` not `charset:utf-8`...